src/dbnode/persist/fs/commitlog/writer.go

```diff
@@ -120,14 +120,27 @@ func newCommitLogWriter(
 ) commitLogWriter {
 	shouldFsync := opts.Strategy() == StrategyWriteWait
 
+	flushSize := opts.FlushSize()
+	flushBufferSize := opts.FlushBufferSize()
+
+	// Allow for at least 2 total buffers.
+	if 2*flushSize > flushBufferSize {
+		flushBufferSize = 2 * flushSize
+	}
```

Comment:
nit: Can we refactor `FlushBufferSize` to be something like `NumBuffers` or something to keep it a multiple of a single buffer size? Then `Validate()` can ensure `FlushBufferSize > 2`